### DISCUSSION TITLE
Python for doctors

### DISCUSSION THREAD

**Discussion by GrigorijSchleifer** [community/community#34830] [Other Feature Feedback, Questions, & Ideas]
Hey folks, gents, ladies, and fellas, I hope you all are safe.

I am a doctor and I learn python in my spare time. Any suggestions on what is the best way to learn a language? I am working through the Python For Everybody track from [Charles R. Severance](https://www.py4e.com/) and playing around with code wars. Not sure where my journey will lead me, but it would be nice to read some stories on how experienced programmers learned python.

Any suggestions on what I should do next after Charles's course? Any projects I should code myself ... 

I am happy with every piece of feedback.

Stay healthy Pythonistas!

**Comment by jfullstackdev** [comment]
in the dev world, there are several specific fields, depending on your target, you want to learn those programming languages involved, but yes, Python is a great starting point if you want to learn programming first,

in the Philippines, mostly they will be looking for a Full Stack, like PHP / Laravel, and you must know the core, like CSS, JavaScript, HTML, AJAX, MySQL etc., so that's what I targeted in the past, I'm mainly focusing on Java / Spring Boot and PHP / Laravel, now I have my full time job in one of those companies

but I did not start as a graduate of IT or any other related course, I'm an Educ graduate, an elementary teacher,

in one of my courses in the past, we had ICT, the instructor introduced us in using the formulas of Excel, I was actually not paying attention until one time, he introduced the IF function and booom ... it caught my interest because I realized it can automate a what would-be manual work, so I said to myself one day, I'll create one program out of curiosity alone

then I had my first job as a teacher in one exclusive university as an elementary teacher, other than teaching duties, there is the non-teaching duties, my team was assigned to monitor the scores of different grade levels up to High School at the Sports Fest, they were really doing it manually and I said, that can be done in Excel automatically, even without prior knowledge how to do that, just the IF that I know in my college course, but I was able to actually create it successfully that I had been assigned the same role for 3 years as I'm maintaining it

after that, I did not stop, I needed to learn a full programming language, it could be VBA as I started using Excel but I learned PureBasic first, it was actually very frustrating and that's the nature of the coding, then Java, again another frustrating experience as if I do not know how to program at all but I never surrendered, then I proceeded to Full Stack, as again that's what mostly is demanded here, so now I have that full time job and a part time College Instructor in the same field

**Comment by VladislavDev** [comment]
Good afternoon!

I think the best way to learn programming is to use it for already known areas of knowledge. 

I'll give you an example. You are a doctor (this is wonderful!) and surely there are tasks for automation in your field. For example, you create template documents every day, and you would like to have an application to fill them out automatically. You can explore the necessary libraries (for example, docs tpl for templating word documents and openpyxl for excel, tkinter for creating a graphical interface). Of course, you need basic knowledge of python, but many courses will suit you here. For in-depth study and applied study, you will have to look for information yourself.

With this approach, you not only learn a programming language, but also solve your daily tasks. New tools can be useful for each task, making your training continuous.

This, of course, is only my opinion, I hope it will help you on your way!

**Reply by perezz281** [reply]
Thanks
